validator/sawtooth_validator/networking/interconnect.py

```diff
@@ -815,6 +1071,35 @@ def send(self, message_type, data, callback=None):
         self._send_receive_thread.send_message(message)
         return fut
 
+    def send_last_message(self, message_type, data, callback=None):
+        """Sends a message of message_type and then close the connection.
+
+        Args:
+            message_type (validator_pb2.Message): enum value
+            data (bytes): serialized protobuf
+            callback (function): a callback function to call when a
+                response to this message is received
+
+        Returns:
+            future.Future
+        """
+        message = validator_pb2.Message(
+            correlation_id=_generate_id(),
+            content=data,
+            message_type=message_type)
+
+        fut = future.Future(message.correlation_id, message.content,
+                            has_callback=True if callback is not None
```

Comment:
We should just pass the callback in to the constructor and get rid of this `has_callback` business.  There are no places where we don't pass the callback in immediately after creating the future.